### PR TITLE
ERC-712: Expanded on reference type encoding

### PR DESCRIPTION
**Pull request by cag**
See https://github.com/MetaMask/eth-sig-util/pull/54 and latest comments in discussion thread.

@Recmo @LogvinovLeon @dekz

### CODE REVIEW

**Comment by eip-automerger** [comment]
Hi! I'm a bot, and I wanted to automerge your PR, but couldn't because of the following issue(s):

 - EIP 712 requires approval from one of (@logvinovleon, remco@wicked.ventures, @dekz)

**Comment by axic** [comment]
@Recmo @LogvinovLeon @dekz can you please approve/reject?

**Comment by axic** [comment]
Ping @Recmo @LogvinovLeon @dekz

File: EIPS/eip-712.md [hunk 1]
```diff
@@ -153,19 +153,25 @@ The `hashStruct` function is defined as
 
 The type of a struct is encoded as `name ‖ "(" ‖ member₁ ‖ "," ‖ member₂ ‖ "," ‖ … ‖ memberₙ ")"` where each member is written as `type ‖ " " ‖ name`. For example, the above `Mail` struct is encoded as `Mail(address from,address to,string contents)`.
 
-If the struct type references other struct types (and these in turn reference even more struct types), then the set of referenced struct types is collected, sorted by name and appended to the encoding. An example encoding is `Transaction(Person from,Person to,Asset tx)Asset(address token,uint256 amount)Person(address wallet,string name)`.
+If the struct type references other struct types (and these in turn reference even more struct types), then the set of referenced struct types is collected, sorted by name and appended to the encoding. An example encoding is `Transaction(Person from,Person to,Asset[] assets)Asset(address token,uint256 amount)Person(address wallet,string name)`.
```
Comment (jflatow) [inline EIPS/eip-712.md]: The example types are not sorted, could this be made more clear?

Comment (cag) [reply EIPS/eip-712.md]: Asset and Person are sorted and appended. The encoded type itself (Transaction) isn't.

**Pull request closed without merging**